src/osd/OSD.cc

```diff
@@ -2205,7 +2205,18 @@ bool OSD::asok_command(string admin_command, cmdmap_t& cmdmap, string format,
       pg->unlock();
     }
     f->close_section();
-  } else {
+  } else if (admin_command == "compact") {
+    dout(1) << "triggering manual compaction" << dendl;
+    utime_t start = ceph_clock_now();
+    store->compact();
```

Comment:
It's a synchronous operation. leveldb/rocksdb's compaction has some impacts to user's request. Giving a warning to user is a better solution. how about :

     WARNING:  compaction possibly slows your requests.